gh-132642: Add documentation on how to render human-readable timedelta object (#133825)
python/cpython: Doc/library/datetime.rst

      >>> (d.days, d.seconds, d.microseconds)
      (-1, 86399, 999999)

>    A :class:`!timedelta` object can be rendered in a human-readable format as follows:

Reviewer: This should be clearer that this is a recipe, this may mislead people with `__str__`.

>    Negative :class:`!timedelta` objects can display in a way that may be confusing. Here's a recipe for displaying both positive and negative `timedelta` objects in a more readable format:

Reviewer 2: ```suggestion
   The string representation of :class:`!timedelta` objects can be confusing,
   but the following recipe can be used to produce a more readable format: 
```

   .. code-block:: pycon

      >>> def pretty_timedelta(td):
      ...     if td.days >= 0:
      ...         return str(td)
      ...     return f'-({-td!s})'
      ...
      >>> d = timedelta(hour=-1)

>       >>> d

Reviewer 2: ```suggestion
      >>> str(d)
```

Sorry, `d` would have used the repr, not the str.

      >>> (d.days, d.seconds, d.microseconds)
      (-1, 86399, 999999)

   The string representation of :class:`!timedelta` objects can be confusing,

>    the following recipe can be used to produce a more readable format:

Reviewer 2: ```suggestion
   Since the string representation of :class:`!timedelta` objects can be confusing,
   use the following recipe to produce a more readable format:
```

I feel the sentence is wrong. We have "X, Y:" and not "Since X, Y" or "X, but you can do Y".

>    but the following recipe can be used to produce a more readable format:

Reviewer: ```suggestion
   the following recipe can be used to produce a more readable format:
```
Not necessary

   .. code-block:: pycon

      >>> def pretty_timedelta(td):
      ...     if td.days >= 0:
      ...         return str(td)
      ...     return f'-({-td!s})'
      ...
      >>> d = timedelta(hours=-1)

>       >>> str(d)

Reviewer: ```suggestion
      >>> str(d)  # not human-friendly
```
Small improvement to the example

      >>> (d.days, d.seconds, d.microseconds)
      (-1, 86399, 999999)

   Negative :class:`!timedelta` objects can display in a way that may be
   confusing. Here's a recipe for displaying both positive and negative

>    :class:`!timedelta` objects in a more readable format:

Reviewer 2: Timedelta objects are not "negative" or positive. I suggest keeping the formulation I suggested.

Author: Indeed! I am going to apply this suggestion https://github.com/python/cpython/pull/133825#discussion_r2083119530
